src/main/java/org/reaktivity/nukleus/oauth/internal/stream/OAuthProxyFactory.java

```diff
@@ -462,8 +604,50 @@ private void onSignal(
                 {
                     writer.doAbort(target, targetRouteId, targetStreamId, traceId, targetAuthorization);
                 }
+            }
+        }
+
+        private boolean tryExtendAuthStateExpiration(
+            int sourceRealmId)
+        {
+            boolean extendedExpiration = false;
+
+            try
+            {
+                final JwtClaims claims = JwtClaims.parse(signature.getPayload());
+                final String subject = claims.getSubject();
+                final Map<String, OAuthAccessGrant> authStateMap = lookupAuthState(sourceRealmId, subject);
+
+                if(authStateMap != null)
+                {
+                    final OAuthAccessGrant authState = authStateMap.get(subject.intern());
```

Comment:
Recommend moving this into `lookup` and returning `null` if `authStateMap` is `null`.